lib/logic/controller.js

```diff
@@ -242,4 +262,26 @@ Controller.prototype._handleMergeError = function(err) {
     );
 };
 
+/**
+ * _while returns a promise which is resolved when condition() is true, repeatedly calling fn()
+ * between checks of condition().
+ */
+Controller.prototype._while = function(condition, fn, opt_delay_ms) {
```

Comment:
This is cute. Perhaps this should be put into a util class? We have one in `/components/utils.js` which should probably be moved up to `/utils.js`.